bam_qc_metrics/bam_qc.py

```diff
@@ -270,10 +384,16 @@ def evaluate_mismatches_by_read(self, input_bam, reference, n_as_mismatch):
         """
         mismatches = {}
         # find read using flags; unknown read is neither R1 nor R2
-        if reference != None:
-            r1_mismatch = self.parse_mismatch(pysam.stats('-r', reference, '-f', '64', input_bam), n_as_mismatch)
-            r2_mismatch = self.parse_mismatch(pysam.stats('-r', reference, '-f', '128', input_bam), n_as_mismatch)
-            ur_mismatch = self.parse_mismatch(pysam.stats('-r', reference, '-F', '192', input_bam), n_as_mismatch)
+        if self.reference != None:
+            r1_mismatch = self.parse_mismatch(pysam.stats('-r', reference,
+                                                          '-f', '64',
+                                                          self.bam_path))
+            r2_mismatch = self.parse_mismatch(pysam.stats('-r', reference,
+                                                          '-f', '128',
+                                                          self.bam_path))
+            ur_mismatch = self.parse_mismatch(pysam.stats('-r', reference,
+                                                          '-F', '192',
+                                                          self.bam_path))
```

Comment:
`reference` is not defined. I think it should be `self.reference`.

Reply:
Right you are. Fixing now. I will also update the tests to use the --reference parameter.

I didn't do that previously because I didn't want to an external dependency on the reference path (human genome reference being too big to commit to the repo) but on reflection I think it's the lesser evil here.

Reply:
`samtools stats` might allow you to use just one chromosome in the `fasta` file. If you pick one of the runt chromosomes (outside of chr1-22, X), it would add less than 1MB to the repo.

Reply:
OK, I think I'll make that a TODO. Right now, for the sake of getting things moving I'll use an external human genome reference path. In the unlikely event someone is running the tests outside of OICR, they can configure the reference themselves.